src/Microsoft.Data/ArrowStringDataFrameColumn.cs

```diff
@@ -469,5 +469,77 @@ protected internal override Delegate GetDataViewGetter(DataViewRowCursor cursor)
 
         private ValueGetter<ReadOnlyMemory<char>> CreateValueGetterDelegate(DataViewRowCursor cursor) =>
             (ref ReadOnlyMemory<char> value) => value = this[cursor.Position].AsMemory();
+
+        public PrimitiveDataFrameColumn<bool> ElementwiseEquals(string value)
+        {
+            ReadOnlySpan<byte> bytes = value != null ? Encoding.UTF8.GetBytes(value) : default(ReadOnlySpan<byte>);
+            PrimitiveDataFrameColumn<bool> ret = new PrimitiveDataFrameColumn<bool>(Name, Length);
+            for (long i = 0; i < Length; i++)
+            {
+                var strBytes = GetBytes(i);
+                if (strBytes.Length != bytes.Length)
+                {
+                    ret[i] = false;
+                    continue;
+                }
+                bool isEqual = true;
+                for (int j = 0; j < bytes.Length; j++)
+                {
+                    isEqual = strBytes[j] == bytes[j];
+                    if (!isEqual)
+                    {
+                        break;
+                    }
+                }
+                ret[i] = isEqual;
+            }
+            return ret;
+        }
+
+        public override PrimitiveDataFrameColumn<bool> ElementwiseEquals<T>(T value)
+        {
+            return ElementwiseEquals(value.ToString());
+        }
+
+        public override PrimitiveDataFrameColumn<bool> ElementwiseEquals(DataFrameColumn column)
+        {
+            return StringDataFrameColumn.ElementwiseEqualsImplementation(this, column);
```

Comment:
This is going to convert every string to a UTF16 string, right? It may be fine for now to get the functionality working, but I think we should log an issue to avoid this when possible. For example, if I call `ElementwiseEquals` on 2 Arrow String columns.

Reply:
Right. I'll log an issue to add an API that takes in an ArrowStringDataFrameColumn as a parameter to avoid the UTF conversions

Reply:
https://github.com/dotnet/corefxlab/issues/2775